codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientHandshaker00.java

```diff
@@ -114,8 +117,40 @@ public WebSocketClientHandshaker00(URI webSocketURL, WebSocketVersion version, S
     WebSocketClientHandshaker00(URI webSocketURL, WebSocketVersion version, String subprotocol,
                                 HttpHeaders customHeaders, int maxFramePayloadLength,
                                 long forceCloseTimeoutMillis, boolean absoluteUpgradeUrl) {
+        this(webSocketURL, version, subprotocol, customHeaders, maxFramePayloadLength, forceCloseTimeoutMillis,
+                absoluteUpgradeUrl, null, null);
+    }
+
+    /**
+     * Creates a new instance with the specified destination WebSocket location and version to initiate.
+     *
+     * @param webSocketURL
+     *            URL for web socket communications. e.g "ws://myhost.com/mypath". Subsequent web socket frames will be
+     *            sent to this URL.
+     * @param version
+     *            Version of web socket specification to use to connect to the server
+     * @param subprotocol
+     *            Sub protocol request sent to the server.
+     * @param customHeaders
+     *            Map of custom headers to add to the client request
+     * @param maxFramePayloadLength
+     *            Maximum length of a frame's payload
+     * @param forceCloseTimeoutMillis
+     *            Close the connection if it was not closed by the server after timeout specified
+     * @param  absoluteUpgradeUrl
+     *            Use an absolute url for the Upgrade request, typically when connecting through an HTTP proxy over
+     *            clear HTTP
+     * @param httpRequestEncoderName
+     *            The name of the HTTP request encoder to be replaced when the handshake is complete
+     * @param httpResponseDecoderName
+     *            The name of the HTTP response decoder to be replaced when the handshake is complete
+     */
+    public WebSocketClientHandshaker00(URI webSocketURL, WebSocketVersion version, String subprotocol,
```

Comment:
should we just make this package-private for now ?

Reply:
Sure, so the idea is to not make it public until a later time when we're reasonably certain that this interface is stable?  That sounds reasonable to me, as long as there is _some_ public way to pass these parameters along.